fitapp/views.py

```diff
@@ -120,15 +120,17 @@ def complete(request):
 def create_fitbit_session(sender, request, user, **kwargs):
     """ If the user is a fitbit user, update the profile in the session. """
 
-    if user.is_authenticated() and utils.is_integrated(user) and \
-            user.is_active:
-        fbuser = UserFitbit.objects.filter(user=user)
-        if fbuser.exists():
-            fb = utils.create_fitbit(**fbuser[0].get_user_data())
-            try:
-                request.session['fitbit_profile'] = fb.user_profile_get()
-            except:
-                pass
+    if user.is_authenticated() and utils.is_integrated(user) and user.is_active:
+        try:
+            fbuser = UserFitbit.objects.get(user=user)
+            fb = utils.create_fitbit(**fbuser.get_user_data())
+            request.session['fitbit_profile'] = fb.user_profile_get()
+            if fb.client.token['access_token'] != fbuser.access_token:
```

Comment:
Of course. If you are doing this anytime in get_fitbit_data why do you decide to not do it here? You need to check it on any connection to fitbit. create_fitbit_session creates connection to fitbit. create_fitbit_session does not run get_fitbit_data. So this check is not running in create_fitbit_session. And you can lose refresh_token because it can be used only once.